site-docs/src/main/paradox/dynamodb/usage/index.md

```diff
@@ -0,0 +1,42 @@
+# DynamoDB Client
+
+The client constructor takes a `DynamoDbAsyncClient` object. The default `apply` method passes one in with sensible defaults but you can pass your own in if necessary.
+
+```scala
+val dynamoDbAsyncClient: DynamoDbAsyncClient = ???
+val clientWithCustom = new DADynamoDBClient(dynamoDbAsyncClient)
+
+val clientWithDefault = DADynamoDBClient()
+```
+
+The client exposes two methods:
+
+```scala
+def getItem(dynamoDbRequest: DynamoDbRequest): F[AttributeValue]
+
+def updateItem(dynamoDbRequest: DynamoDbRequest): F[Int]
+```
+
+The method takes a dynamoDbRequest Case Class:
+
+```scala
+  case class DynamoDbRequest(tableName: String,
+                             primaryKeyAndItsValue: Map[String, AttributeValue],
+                             itemName: String,
+                             itemValueToUpdate: Option[AttributeValue]=None)
+```
+consisting of:
+- tableName - the DynamoDB table you want to make the edits in
+- primaryKeyAndItsValue - the primary key and the value to filter it down
+  - you can use the .builder and pass the value to the 'type' method ('.s', '.b', '.n' etc)
+  - *Warning* adding a value that isn't already present in the table will cause DynamoDb to add another row so watch out for misspellings
+- itemName - the item name you want the value of
+- itemValueToUpdate - the new value you want to take place of the current one 
```

Comment:
+ And other DDB features like atomic counters? https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.AtomicCounters

Reply:
Regarding the update/add multiple attributes, I think the idea for these clients is that we are just going to create methods for operations that we need right now and are just going to leave it for the next person that needs additional functionality to implement it in the future. Also, from what I can see, there is only a `BatchGetItem` and not a `BatchUpdateItem` operation.

Regarding the atomic counter, I'm not sure how to implement this; there isn't a lot of information out there about it.

Reply:
My concern is that by starting with a method that allows only 1 attribute to be fetched or updated at a time we'll need to define new methods or change the way existing methods work as we develop, that in turn means changing the way our business logic calls these methods. For example, if a ddb item had 2 attributes and we wanted to change both of these we wouldn't want to call `updateAttributeValue` twice

Maybe it's worth changing these methods to be attribute agnostic...
- getItem to return all item attributes by default and support a subset of attributes when passed as an argument
- updateItem to expect a map of attribute->value instead of a single attribute+value

I think for now we can ignore atomic counter until we need to use it but it is an interesting use case if we wanted to increment the counter and change other attributes at the same time.